bigqueue_flush_test.go

```diff
@@ -169,37 +160,42 @@ func TestResetOfBothFlushIntervalStates(t *testing.T) {
 	if err := bq.Enqueue(msg); err != nil {
 		t.Fatalf("enqueue failed :: %v", err)
 	}
+	checkMutOps(bq, t, 1)
 
-	checkMutateOps(bq, t, 1)
-	clock.Advance(flushElapsedDuration) // time interval reached, flush happens, resets both prevFlushTime, mutateOps
+	// time interval reached, flush happens, resets both lastFlush, mutOps
+	clock.Advance(flushPeriod)
 
 	if err := bq.Enqueue(msg); err != nil {
 		t.Fatalf("enqueue failed :: %v", err)
 	}
+	checkMutOps(bq, t, 0)
+	checkLastFlushTime(bq, t, clock.Now())
 
-	checkMutateOps(bq, t, 0)
-	prevFlushTime := (bq.(*MmapQueue)).prevFlushTime
-	if prevFlushTime != clock.Now() {
-		t.Fatalf("expected prevFlushTime: %v to be reset to clock's now: %v", prevFlushTime, clock.Now())
-	}
-
-	clock.Advance(flushElapsedDuration / 2) // time moves ahead
+	// time moves ahead by half of flush period
+	clock.Advance(flushPeriod / 2)
 
+	// flush after 2 mutOps, causes mutOps reset and time reset to Now()
 	if err := bq.Enqueue(msg); err != nil {
 		t.Fatalf("enqueue failed :: %v", err)
 	}
 	if err := bq.Enqueue(msg); err != nil {
 		t.Fatalf("enqueue failed :: %v", err)
 	}
+	checkMutOps(bq, t, 0)
+	checkLastFlushTime(bq, t, clock.Now())
 
-	// flush after 2 mutateOps, causes mutateOps reset and time reset to Now()
-
-	checkMutateOps(bq, t, 0)
-	prevFlushTime = (bq.(*MmapQueue)).prevFlushTime
-	if prevFlushTime != clock.Now() {
-		t.Fatalf("expected prevFlushTime: %v to be reset to clock's now: %v", prevFlushTime, clock.Now())
+	// move time ahead and mutate some ops
+	clock.Advance(flushPeriod)
+	if err := bq.Enqueue(msg); err != nil {
+		t.Fatalf("enqueue failed :: %v", err)
 	}
+	checkMutOps(bq, t, 0)
+	checkLastFlushTime(bq, t, clock.Now())
 
+	clock.Advance(flushPeriod)
```

Comment:
advance only by flushPeriod/2 or something less?
to assert that bq.Flush() is an unconditional flush.